hw2/04-background.js

```diff
@@ -1 +1,64 @@
 // Add your code here
+// Function to generate a random color in rgba format with a dimmed alpha value
+function generateRandomColor() {
+    const r = Math.floor(Math.random() * 256);
+    const g = Math.floor(Math.random() * 256);
+    const b = Math.floor(Math.random() * 256);
+    const alpha = 0.5; // Dimmed alpha value
+    return `rgba(${r}, ${g}, ${b}, ${alpha})`;
+  }
+  
+  let intervalId;
+  let intervalSeconds = 3; // Default interval is 3 seconds
+  let isRunning = true; // Flag to track if color change is running or stopped
+  // Function to start changing background color at the specified interval
+  function startBackgroundColorChange() {
+    intervalId = setInterval(function() {
+      document.body.style.backgroundColor = generateRandomColor();
+    }, intervalSeconds * 1000);
+  }
+  
+  // Function to stop changing background color
+  function stopBackgroundColorChange() {
+    clearInterval(intervalId);
+  }
+  
+  // Function to handle interval input change
+  function handleIntervalInputChange(event) {
+    intervalSeconds = parseInt(event.target.value);
+    if (!isNaN(intervalSeconds) && intervalSeconds > 0  && !isRunning) {
+      stopBackgroundColorChange(); // Stop current interval
+      startBackgroundColorChange(); // Start with new interval
+    }
+  }
+  
+  // Initialize the background color change
+  startBackgroundColorChange();
```

Comment:
Their could be other ways to handle color change, event listeners added to id elements could be extremely effective in cases like these.